src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -13,8 +13,9 @@ namespace Microsoft.CodeAnalysis.CSharp
     internal class ExpressionVariableFinder : CSharpSyntaxWalker
     {
         private Binder _scopeBinder;
-        private Binder _enclosingBinderOpt;
+        private Binder _enclosingBinder;
         private ArrayBuilder<LocalSymbol> _localsBuilder;
+        private SyntaxNode _nodeToBind;
 
         internal static void FindExpressionVariables(
```

Comment:
>FindExpressionVariables [](start = 29, length = 23)

It is hard to verify that the visitor is properly handling _nodeToBind for every node that can pass through here. It would be good to have an assert for all non-expression nodes that we expect to to see. This should also help us in the future as new constructs that require special handling are added to the language.